src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java

```diff
@@ -35,33 +31,70 @@ public class VaultHelper {
     static Map<String, String> getVaultSecret(@NonNull String secretPath,
                                               @CheckForNull String prefixPath,
                                               @CheckForNull String namespace,
-                                              @CheckForNull Integer engineVersion,
-                                              @NonNull ItemGroup<Item> context) {
+                                              @CheckForNull Integer engineVersion) {
+        VaultConfiguration configuration = null;
+        ItemGroup jenkins = Jenkins.get();
```

Comment:
See https://github.com/jenkinsci/hashicorp-vault-plugin/issues/201#issuecomment-1109723672

Reply:
I think this would be fine if `getVaultSecret` is now only ever called on the controller. Getting rid of the `SlaveToMasterCallable` should definitely be a priority. CC @daniel-beck